### ISSUE TITLE
Invalidate schema cache when new collection/field is accessed through `/collections` / `/fields`

### ISSUE THREAD

**Issue by enzorrr** [directus/directus#6602]
### Describe the Bug

Hello,

I'm struggling to map existing tables to directus collections.
When creating collections from the Admin-App everything works fine, but when I have "Database only" Tables, I always run into `[FORBIDDEN] You don't have permission to access this.` when i click on "Database only: Click to Configure".

![error](https://user-images.githubusercontent.com/1351948/124143751-acd6ce00-da8b-11eb-8770-fea4304ff036.PNG)

In Chrome's Network panel  I can see the following request, returning 403:

![request](https://user-images.githubusercontent.com/1351948/124143413-62ede800-da8b-11eb-94f0-abc97c49f4a1.PNG)

The entry in directus_collections is made, so I can see the collection in the Admin App., but allways geht the forbidden error. No matter if I am admin or grant all permissions to public and try to get it via api.

Am I missing something here? 

### To Reproduce

Reproduce:
 
1. Run a working directus installation
2. Create a table directly on the database
3. Go to the Admin App > Settings > Datamodel
4. Klick the unconfigured Table
5. Klick a field to configure it
6. The configuration view opens, but the forbidden-message is already popping up in the background!

Tested on rc.83/postgresql and rc.48/mysql

### What version of Directus are you using?

9.0.0 rc.45 / rc.83

### What version of Node.js are you using?

14.17

### What database are you using?

posgresql 12.7 / mysql 5.7

### What browser are you using?

Chrome

### What operating system are you using?

Ubuntu 20.04

### How are you deploying Directus?

running locally

**Comment by enzorrr** [comment]
addition:
I also cannot delete these collections (same error) whereas I can delete collections I created from within the admin app wihout a flaw.

**Comment by rijkvanzanten** [comment]
Just to make sure, when you said version `9.0.0 rc.45 / rc.83`. Are these fresh installations, or was 45 updated to 83? If the latter, did you run the database migrations when updating?

**Comment by rgieseke** [comment]
Is there something in your Directus logs? I ran into the same error for a table without a Primary Key.

Something like
11:46:28 ⚠️  Collection "..." doesn't have a primary key column and will be ignored

The table was shown in the Admin interface but otherwise same error as yours.

**Comment by enzorrr** [comment]
> Just to make sure, when you said version `9.0.0 rc.45 / rc.83`. Are these fresh installations, or was 45 updated to 83? If the latter, did you run the database migrations when updating?

both separate fresh installations.

**Comment by enzorrr** [comment]
> Is there something in your Directus logs? I ran into the same error for a table without a Primary Key.
> 
> Something like
> 11:46:28 ⚠️ Collection "..." doesn't have a primary key column and will be ignored
> 
> The table was shown in the Admin interface but otherwise same error as yours.

Didn't see something like that in the logs (loglevel=trace), maybe I've overseen it?! But with just a quick try, it seems to resolve the issue in the mysql installation running on r.45.

However things are still strange when using postgres. Some Tabels suddenly were working. No change and no server-restart in between... even before I read your comment with the primary key.... I just logged on the other day and bang... working 😀 .... but still not all of them (all with a Primary Key) 
Will dig a little deeper in it today and try to debug whats going on.

**Comment by enzorrr** [comment]
Got the reason. I guess I was too impatient and expected things to work immediately after I executed the "CREATE TABLE...." statement.
The 403 came from `middleware/collection-exists.ts:14` as `req.schema.collections` was not up to date.
I then looked for actions where the schemaCache is invalidated in `services/collections.ts` and everything worked after performing one of these actions. (That is why it seemed so random to me in the beginning, when it suddenly started working...)
Probably wouldn't have been an issue if I did some more things in the app, constantly hitting "cache.clear()" methods, but my only focus was to create tables in sql and then add them in directus.
Would be a nice thing to see that the schemaCache is invalidated whenever a new table is recognized, but so far I know how to overcome this. 
Thanks for your help 👍
